apps/air-discount-scheme/backend/src/app/modules/user/user.controller.ts

```diff
@@ -79,13 +79,22 @@ export class PrivateUserController {
         '[/relations] Request parameters do not correspond with user authentication.',
       )
     }
-    const relations = await this.userService.getRelations(params.nationalId)
-    const users = (await Promise.all([
-      this.userService.getUserInfoByNationalId(params.nationalId),
-      ...relations.map((nationalId) =>
-        this.userService.getUserInfoByNationalId(nationalId),
-      ),
-    ])) as User[]
-    return users.filter((user) => user) as User[]
+    const relations = await this.userService.getRelations(authUser)
+
+    // Adding User in beginning of array so order is correct.
+    relations.unshift(authUser.nationalId)
+    const userAndRelatives = await this.userService.getMultipleUsersByNationalIdArray(
+      relations,
+    )
+
+    const returnUsers = userAndRelatives.filter(
```

Comment:
hopefully with the other suggested changes, you don't have to perform this filter statement, and type coercion would be necessary

Reply:
Since getUserInfoByNationalId can return either User or null - depending on the data of the server(not a problem in production 99.9999999999% of the time) then I believe the filter statement will be required for dev/staging and the most extreme edgecase.
This filter statement could be moved into getMultipleUserByNationalArray but I feel it is more appropriate that the controller handles what is legit and what isn't.
Feel free to voice your opinion on that

Reply:
not really an opinion, but like i said.. moving the filter statement to the other method would fix that

Reply:
I fixed it - earlier I missed two other comments.